scripts/fixture-metadata-hashes

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+
+import hashlib
+import re
+from pathlib import Path
+from typing import List
+
+# This especially terrible html "parsing" is intended to roundtrip with edits.
+TERRIBLE_ANCHOR_RE = re.compile(r"(\s*<a)(.*?)(/?>.*)", re.S)
+TERRIBLE_ATTR_RE = re.compile(r'([\w-]+)(=)"(.*?)"')
+
+
+class NotAnAnchor(Exception):
+    pass
+
+
+class Anchor:
+    def __init__(self, text: str) -> None:
+        m = TERRIBLE_ANCHOR_RE.match(text)
+        if m is None:
+            raise NotAnAnchor()
+        self.header = m.group(1)
+        self.trailer = m.group(3)
+
+        self.attrs: Dict[str, str] = {}
+        for attr in TERRIBLE_ATTR_RE.finditer(m.group(2)):
+            self.attrs[attr.group(1)] = attr.group(3)
+
+    def __str__(self):
+        buf: List[str] = []
+        buf.append(self.header)
+        for k, v in self.attrs.items():
+            buf.append(f' {k}="{v}"')
+        buf.append(self.trailer)
+        return "".join(buf)
+
+
+def main():
+    FIXTURE_BASE = Path("hdeps/tests/fixtures/")
+    for p in FIXTURE_BASE.glob("*.html"):
```

Comment:
Minor: all uppercase probably not needed (and some linter rule is probably going to complain about it if enabled)